src/sentry/api/bases/external_actor.py

```diff
@@ -0,0 +1,120 @@
+from typing import Any, MutableMapping
+
+from django.db import IntegrityError
+from django.http import Http404
+from rest_framework import serializers
+from rest_framework.exceptions import PermissionDenied
+
+from sentry import features
+from sentry.api.serializers.rest_framework.base import CamelSnakeModelSerializer
+from sentry.api.validators.integrations import validate_provider
+from sentry.models import ExternalActor, Organization, OrganizationMember, Team, User
+from sentry.types.integrations import EXTERNAL_PROVIDERS
+
+
+class ExternalActorSerializerBase(CamelSnakeModelSerializer):  # type: ignore
+    external_id = serializers.CharField()
+    external_name = serializers.CharField(required=True)
+    provider = serializers.ChoiceField(choices=list(EXTERNAL_PROVIDERS.values()))
+
+    @property
+    def organization(self):
+        return self.context["organization"]
+
+    def get_actor_id(self, validated_data: MutableMapping[str, Any]) -> int:
+        target_option = validated_data.pop(self._actor_key, None)
+
+        if not target_option:
+            raise Exception("Invalid actor")
+
+        return target_option.actor_id
+
+    def get_provider_id(self, validated_data: MutableMapping[str, Any]) -> int:
+        provider_name_option = validated_data.pop("provider", None)
+        return validate_provider(provider_name_option).value
```

Comment:
```suggestion
validate_provider(provider_name_option, available_providers=["github", "gitlab"])
```

Reply:
maybe we can set `["github", "gitlab"]` as a constant somewhere so there's only one place to update the `available_providers`